beluga_ros/src/amcl.cpp

```diff
@@ -50,6 +50,20 @@ void Amcl::update_map(beluga_ros::OccupancyGrid map) {
   std::visit([&](auto& sensor_model) { sensor_model.update_map(std::move(map)); }, sensor_model_);
 }
 
+void Amcl::update_propagation(Sophus::SE2d base_pose_in_odom) {
+  if (particles_.empty()) {
+    return;
+  }
+
+  // Force propagation without checking update_policy_
+  std::visit(
+      [&, this](auto& policy, auto& motion_model) {
+        particles_ |= beluga::actions::propagate(policy, motion_model(control_action_window_ << base_pose_in_odom)) |
+                      beluga::actions::normalize(policy);
```

Comment:
We don't need to normalize, since we only affected distribution density, but we left the weights unchanged.

Reply:
+1